test/molecule-role/molecule/default/prepare.yml

```diff
@@ -92,29 +96,52 @@
     - name: Add stackstate-agent Apt signing key
       become: true
       apt_key:
-        url: https://s3.amazonaws.com/stackstate-agent-test/gpg
+        url: "{{agent_repo_url}}/public.key"
         state: present
+    - name: Inform about current branch
+      debug: msg="deb {{agent_repo_url}} {{agent_current_branch}} main"
     - name: Add stackstate-agent Apt repository
       become: true
       apt_repository:
-        repo: deb https://s3.amazonaws.com/stackstate-agent-test master main
+        repo: "deb {{agent_repo_url}} {{agent_current_branch}} main"
     - name: Install StackState-Agent
       become: true
       apt:
         name: stackstate-agent
         state: present
         update_cache: yes
-    - name: Configure StackState-Agent
-      copy:
-        content: |
-          # The host of the Datadog intake server to send Agent data to
-          sts_url: http://{{ hostvars['instance-2']['ansible_host'] }}:7077/stsAgent
-          api_key: API_KEY
-        dest: /etc/stackstate-agent/stackstate.yaml
+    - name: Copy stackstate config file
+      become: true
+      command: cp -a /etc/stackstate-agent/stackstate.yaml.example /etc/stackstate-agent/stackstate.yaml
+    - name: Replace sts_url
       become: true
+      lineinfile:
+        path: /etc/stackstate-agent/stackstate.yaml
+        regexp: '^sts_url:'
+        line: "sts_url: http://{{ hostvars['receiver']['ansible_host'] }}:7077/stsAgent"
+    - name: Replace process_sts_url
+      become: true
+      lineinfile:
+        path: /etc/stackstate-agent/stackstate.yaml
+        regexp: '^  process_sts_url:.*'
+        line: "  process_sts_url: http://{{ hostvars['receiver']['ansible_host'] }}:7077/stsAgent"
     - name: Run stackstate-agent
       become: true
       systemd:
         name: stackstate-agent
         state: started
         daemon_reload: yes
+- name: Prepare Agent 1 listening
+  hosts: agent1
+  gather_facts: false
+  tasks:
+    - name: Open listening port on agent 1
+      shell: nohup nc -l -p 125 > /dev/null &
+      become: true
```

Comment:
if we use a higher port than 125 you don't need `become`, right ?

Reply:
Higher ports could be already occupied,  so i'd rather use a low one